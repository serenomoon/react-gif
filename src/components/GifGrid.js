import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


  const {data:images, loading} = useFetchGifs(category);

  


  return (
    <>
      <h2 className='animate__animated animate__fadeIn'>{category}</h2>

      {loading && <p className='animate__animated animate__flash'>loading...</p>}

      <div className='card-grid'>

            {
              images.map(img => ( 
                <GifGridItem 
                    key={img.id}
                    {...img}
                  />
              ))
            }

      </div>
    </>
  );
};

GifGrid.propTypes = {
  category : PropTypes.string.isRequired
}
