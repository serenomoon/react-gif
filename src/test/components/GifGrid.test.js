import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba <GifGrid />', () => {
    
    const category = 'Hola Mundo'
    

    test('Debe mostrar correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id : '123',
            title : 'Cualquie cosa',
            url: 'http://localhost/cualquiercosa.jpg'
        },
        {
            id : '456',
            title : 'Cualquie cosa 2',
            url: 'http://localhost/cualquiercosa2.jpg'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category }/>);

        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });

});