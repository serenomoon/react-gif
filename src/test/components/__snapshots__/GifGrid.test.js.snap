// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGrid /> Debe mostrar correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hola Mundo
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
